src/core.js

```diff
@@ -3,6 +3,7 @@ var matched, browser,
 	oldInit = jQuery.fn.init,
 	oldParseJSON = jQuery.parseJSON,
 	rspaceAngle = /^\s*</,
+	rattrHash = /\[\s*\w+\s*[~|^$*]?=\s*#/,
```

Comment:
We can be even more sure that nothing invalid falls through the cracks (e.g., `"[attr=foo#bar]"`) by inserting `(?![\s'"])[^#\]]*` before the final `#`.